File: backend/src/models/page.ts
import mongoose from "mongoose";

const Schema = mongoose.Schema

let Page = new Schema({
    name: {
        type: String
    },
    headline: {
        type: String
    },
    industry: {
        type: String
    },
    location: {
        type: String
    },
    profile_image: {
        type: String
    },
    about: {
        type: String
    },
    website: {
        type: String
    },
    date_created: {
        type: Number
    },
    admin: {
        type: String
    },
    specialties: {
        type: String
    },
    followers: [
        {
            user: {
                type: String
            }
        }
    ],
    reviews: [
        {
            user: {
                type: String
            },
            text: {
                type: String
            },
            user_image: {
                type: String
            }
        }
    ]
})

export default mongoose.model('Page', Page, 'page')

File: WorkHub/app/src/main/java/com/example/workhub/data/retrofit/models/Page.kt
package com.example.workhub.data.retrofit.models

data class Page(
    val name: String,
    val headline: String,
    val location: String,
    val profile_image: String,
    val about: String,
    val website: String,
    val date_created: Long,
    val specialties: String,
    val admin: String,
    var followers: List<Follower>,
    var reviews: List<PageReview>
)

data class Follower(
    val user: String
)

data class PageReview(
    val user: String,
    val text: String,
    val user_image: String
)
